arangod/Graph/KShortestPathsFinder.cpp

```diff
@@ -76,12 +80,41 @@ void KShortestPathsFinder::clear() {
   _traversalDone = true;
 }
 
+#ifdef USE_ENTERPRISE
+bool KShortestPathsFinder::pathContainsOnlySatellites() const {
+  // Must be called after path calculation is finished.
+  if (options().isDisjoint() && _smartValue.empty()) {
+    return true;
+  }
+  return false;
+}
+#endif
+
 // Sets up k-shortest-paths traversal from start to end
 bool KShortestPathsFinder::startKShortestPathsTraversal(
     arangodb::velocypack::Slice const& start,
     arangodb::velocypack::Slice const& end) {
   TRI_ASSERT(start.isString());
   TRI_ASSERT(end.isString());
+
+#ifdef USE_ENTERPRISE
+  if (options().isDisjoint()) {
```

Comment:
This piece of code repeats, is it possible (without too much overhead) to have it once?